Content.Server/Nutrition/EntitySystems/FoodSystem.cs

```diff
@@ -198,10 +199,10 @@ private void DeleteAndSpawnTrash(EntityUid userUid, FoodComponent component)
         //TODO: DoAfter based on delay after food & drinks delay PR merged...
         private void AddEatVerb(EntityUid uid, FoodComponent component, GetInteractionVerbsEvent ev)
         {
-            Logger.DebugS("action", "triggered");
             if (!ev.CanInteract ||
                 !EntityManager.TryGetComponent(ev.User.Uid, out SharedBodyComponent? body) ||
-                !_bodySystem.TryGetComponentsOnMechanisms<StomachComponent>(ev.User.Uid, out var stomachs, body))
+                !_bodySystem.TryGetComponentsOnMechanisms<StomachComponent>(ev.User.Uid, out var stomachs, body) ||
+                !ev.User.Uid.InRangeUnobstructed(uid, popup: false))
                 return;
```

Comment:
You should be able to just use `ev.CanAccess`. Because many verbs need to check `InRangeUnobstucted` it's just sent as an event arg so that you don't need to have every system check it individually.